sunpy/data/data_manager/cache.py

```diff
@@ -0,0 +1,139 @@
+import os
+from pathlib import Path
+from datetime import datetime, timedelta
+from warnings import warn
+from urllib.request import urlopen
+
+import astropy.units as u
+from astropy.time import TimeDelta
+
+from sunpy.data.data_manager.downloader import DownloaderError
+from sunpy.util.net import get_filename
+from sunpy.util.util import hash_file, replacement_filename
+
+
+class Cache:
+    """
+    Cache handles caching.
+
+    Parameters
+    ----------
+    downloader: Implementaion of `~sunpy.data.data_manager.downloader.DownloaderBase`
+        Downloader object for downloading remote files.
+    storage: Implementaion of `~sunpy.data.data_manager.storage.StorageProviderBase`
+        Storage to store metadata about the files.
+    cache_dir: `str` or `pathlib.Path`
+        Directory where the downloaded files will be stored.
+    expiry: `astropy.units.quantity.Quantity` or `None`
+        The interval after which the cache is invalidated. If the expiry is `None`,
+        then the expiry is not checked (or the cache never expires).
+    """
+
+    def __init__(self, downloader, storage, cache_dir, expiry=10*u.s):
+        self._downloader = downloader
+        self._storage = storage
+        self._cache_dir = Path(cache_dir)
+        self._expiry = expiry if expiry is None else TimeDelta(expiry)
```

Comment:
Why not just set it to `TimeDelta(expiry)`?

Reply:
Or is None checked below?

Reply:
`TimeDelta(None)` doesn't work(?).

Reply:
I mean why do you allow None?

Reply:
It's for disabling the expiry check.

Reply:
Oh ok. Could that be in the parameter description please?